src/threading/Manager.cc

```diff
@@ -1,33 +1,45 @@
+#include <sys/socket.h>
+#include <unistd.h>
 
 #include "Manager.h"
 #include "NetVar.h"
+#include "iosource/Manager.h"
 
 using namespace threading;
 
+void HeartbeatTimer::Dispatch(double t, int is_expire)
+	{
+	if ( is_expire )
+		return;
+
+	thread_mgr->SendHeartbeats();
+	thread_mgr->StartHeartbeatTimer();
+	}
+
 Manager::Manager()
 	{
 	DBG_LOG(DBG_THREADING, "Creating thread manager ...");
 
 	did_process = true;
 	next_beat = 0;
-	terminating = false;
-	SetIdle(true);
 	}
 
 Manager::~Manager()
 	{
 	if ( all_threads.size() )
 		Terminate();
+
```

Comment:
use smart pointer?

Reply:
Same problem as the other one. We eventually call TimerMgr::Add() which takes a raw pointer, which inserts the raw pointer into a `PriorityQueue`. I could pass the raw pointer out of the smart pointer, but that kind of defeats the purpose.

Reply:
I'll make a note to make a pass at converting timers to be shared pointers, but not during this work.